server/app/services/cloud/websocket_client.rb

```diff
@@ -195,5 +263,22 @@ def verify_connection
     rescue => exc
       error exc.message
     end
+
+    # Abort the connection, closing the websocket, with a timeout
+    def close
```

Comment:
This copy-pasting of the websocket-rpc related code between the server and agent in the same git repo is really starting to bug me.

Couldn't we somehow refactor these into shared code between the two components?